src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java

```diff
@@ -111,9 +111,8 @@ public static IToken assignRequest(final IStandardRequestManager manager, final
      *
      * @throws IllegalArgumentException is thrown when the request is unknown to this manager.
      */
-    @SuppressWarnings(Suppression.UNCHECKED)
-    public static IToken assignRequestDefault(final IStandardRequestManager manager, final IRequest request, final Collection<IToken> resolverTokenBlackList)
-      throws IllegalArgumentException
+    @SuppressWarnings({UNCHECKED,RAWTYPES})
+    public static IToken<?> assignRequestDefault(final IStandardRequestManager manager, final IRequest request, final Collection<IToken<?>> resolverTokenBlackList)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove usage of generic wildcard type. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 28 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)